src/beast/site_scons/site_tools/VSProject.py

```diff
@@ -583,6 +586,43 @@ def _guid(seed, name = None):
     guid = "{%s-%s-%s-%s-%s}" % (d[:8], d[8:12], d[12:16], d[16:20], d[20:32])
     return guid
 
+def _rmExeMod(file_name):
+    if platform.system() != 'Windows':
+        st = os.stat(file_name)
+        os.chmod(file_name, st.st_mode & ~0111)
+        return
+    
+    import win32security
+    import ntsecuritycon
+
+    user, domain, type = win32security.LookupAccountName ("", "Everyone")
+
+    sd = win32security.GetFileSecurity(file_name, win32security.DACL_SECURITY_INFORMATION)
+    dacl = sd.GetSecurityDescriptorDacl()
+
+    deniedAccess = ntsecuritycon.FILE_EXECUTE
+    dacl.AddAccessDeniedAce(win32security.ACL_REVISION, deniedAccess, user)
+
+    sd.SetSecurityDescriptorDacl(1, dacl, 0)
+    win32security.SetFileSecurity(file_name, win32security.DACL_SECURITY_INFORMATION, sd)
+
+@contextlib.contextmanager
+def openNoExe(file_name):
+    try:
+        f = open(file_name, 'wb')
+    except IOError, detail:
+        raise SCons.Errors.InternalError('Unable to open "' +
+                                         str(file_name) + '" for writing:' + str(detail))
+    try:
+        yield f
```

Comment:
The try is needed because the block that uses the context manager may throw. Consider the how the context manager would act without the try block:
```
from contextlib import contextmanager

@contextmanager
def foo(a):
    yield a
    print("Clean up!")
    
def bar(a):
    raise Exception("Blech!")
    
def baz():
    with foo("a") as a:
        bar(a)
```
Cleanup is never called. I need the try/finally to make sure cleanup is called in all cases.